internal/host/hostcommands/verify_vips_cmd.go

```diff
@@ -0,0 +1,81 @@
+package hostcommands
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type verifyVipsCmd struct {
+	baseCmd
+	db *gorm.DB
+}
+
+func newVerifyVipsCmd(log logrus.FieldLogger, db *gorm.DB) CommandGetter {
+	return &verifyVipsCmd{
+		baseCmd: baseCmd{log: log},
+		db:      db,
+	}
+}
+
+func (f *verifyVipsCmd) prepareParam(host *models.Host) (string, error) {
+	var inventory models.Inventory
+	err := json.Unmarshal([]byte(host.Inventory), &inventory)
+	if err != nil {
+		f.log.WithError(err).Warn("Inventory parse")
+		return "", err
+	}
+	if host.ClusterID == nil || *host.ClusterID == "" {
+		return "", nil
+	}
+	cluster, err := common.GetClusterFromDB(common.LoadTableFromDB(common.LoadTableFromDB(f.db, common.APIVIPsTable), common.IngressVIPsTable),
+		*host.ClusterID, common.SkipEagerLoading)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get cluster %s", host.ClusterID.String())
+	}
+	if len(cluster.APIVips) == 0 && len(cluster.IngressVips) == 0 {
+		return "", nil
+	}
+	request := models.VerifyVipsRequest{}
+	for _, vip := range cluster.APIVips {
```

Comment:
might look a little cleaner with funk.ForEach

Reply:
I am not sure I agree. It needs to modify external variable request, so the append will be in the ForEach function parameter.
So the ForEach will look like
```
funk.ForEach(cluster.APIVips, func(vip *models.APIVip) {request = append(request, &models.VerifyVip{Vip:     vip.IP,  VipType: models.VipTypeAPI})})
```
Is it cleaner ?

Reply:
Apparently that's exactly the example in gofunk.[ForEachRight](https://github.com/thoas/go-funk#funkforeachright):) I think it looks a bit cleaner as we don't need the key inside the loop. But don't have a strong opinion on it.

Reply:
To me seems to complicate.  Prefer to skip.